cmd/krew/cmd/root.go

```diff
@@ -209,11 +210,21 @@ func cleanupStaleKrewInstallations() error {
 }
 
 func checkIndex(_ *cobra.Command, _ []string) error {
-	if ok, err := gitutil.IsGitCloned(paths.IndexPath(constants.DefaultIndexName)); err != nil {
-		return errors.Wrap(err, "failed to check local index git repository")
-	} else if !ok {
+	entries, err := ioutil.ReadDir(paths.IndexBase())
+	if err != nil {
+		return errors.Wrap(err, "failed to list directory")
+	}
+	if len(entries) == 0 {
 		return errors.New(`krew local plugin index is not initialized (run "kubectl krew update")`)
 	}
+	for _, e := range entries {
+		indexPath := paths.IndexPath(e.Name())
+		if ok, err := gitutil.IsGitCloned(indexPath); err != nil {
+			return errors.Wrap(err, "failed to check local index git repository")
+		} else if !ok {
+			return errors.New(`krew local plugin index is not initialized (run "kubectl krew update")`)
```

Comment:
I think this error message needs to be updated but I'm not exactly sure to what. if we've hit this case that means that there was a non git directory in `paths.IndexBase()`. my first instinct is to probably fail if we encounter this since we don't want users manually messing around with their krew folders, krew should take care of that. maybe an error like `non git directory found in index folder`? although I think that's somewhat cryptic.

cc @ahmetb @corneliusweig for your inputs on this.

Reply:
Can we just issue a warning to remind users? like "Invalid index directory"?

Reply:
sorry this fell off my radar.

I think maybe "invalid index found" is ok. My main problem with this is that there's no additional info for the user if they run into this case. Although I think this is most likely ok since if a user runs into this that means they either manually tampered with their krew directory or there's a bug in krew. If its the latter then hopefully an issue would be filed alerting us.

Reply:
@chriskim06 Thanks your reply, I updated error message.  Please take a look.